### PR TITLE
Feature/datepicker remove confirm cancel buttons reworked

### PR DESCRIPTION
**Pull request by markus-walther**
This is an alternative version of PR #1708.

It tries to improve code clarity, but remain feature compatible.

### CODE REVIEW

File: src/components/20-molecules/datepicker/index.js [hunk 1]
```diff
@@ -87,6 +88,35 @@ const parseAndFormatAllowedYears = (allowedyears = [], setYear) => {
...
+
+  if (typeof year === 'number' && year >= 0) {
+    date.setFullYear(year);
+  }
+
+  if (typeof month === 'number' && month >= 0) {
+    date.setMonth(month);
+    // month not changed as desired since target day unavailable?
+    // (e.g. July 31 =/=> June 31)
+    if (date.getMonth() !== month) {
```
Comment (raphaellueckl) [inline src/components/20-molecules/datepicker/index.js:103]: I don't understand. This IF statement seems obsolete to me?

File: src/components/20-molecules/datepicker/index.js [hunk 2]
```diff
@@ -278,18 +309,21 @@ class AXADatepicker extends NoShadowDOM {
     style.width = getFormattedStyle(width); // set width to component's css
     const formattedHeight = getFormattedStyle(height); // set height to input-wrap element because of optional label
 
-    const cellClasses = ({ sameMonth, today, selected, inactive }) =>
-      classMap({
+    const cellClasses = ({ sameMonth, today, selected, inactive, value }) => {
```
Comment (raphaellueckl) [inline src/components/20-molecules/datepicker/index.js]: A comment would imho be helpful about what defines a selected cell. Somehow either if it is user selected, if it is preselected AND if it's 'today', right?

File: src/components/20-molecules/datepicker/index.js [hunk 3]
```diff
@@ -444,14 +478,13 @@ class AXADatepicker extends NoShadowDOM {
...
     }
 
-    if (!this.isControlled) {
-      this.initDate(
-        null,
-        year || startDate.getFullYear(),
-        month || startDate.getMonth(),
-        day || startDate.getDate()
-      );
+    if (this.isControlled) {
```
Comment (raphaellueckl) [inline src/components/20-molecules/datepicker/index.js:480]: Easier readable would imho be: 

```
if (!this.isControlled) {
  // Rest of the method...
}
```

But this is just personal preference.

File: src/components/20-molecules/datepicker/index.js (hunk 2 above)
Comment (markus-walther) [reply src/components/20-molecules/datepicker/index.js]: OK, will do. It's selected iff the same day as the internal this._date.

File: src/components/20-molecules/datepicker/index.js (hunk 3 above)
Comment (markus-walther) [reply src/components/20-molecules/datepicker/index.js:480]: This is what I had before, as the diff shows. The disadvantage is more nesting/indentation, which is usually harder to read. As you say, it's a preference, but those were my reasons.

File: src/components/20-molecules/datepicker/index.js (hunk 2 above)
Comment (markus-walther) [reply src/components/20-molecules/datepicker/index.js]: Even better action: remove reference to selected here, it's useless ;-)

**Pull request merged**